go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -472,34 +487,30 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 			return fmt.Errorf("error when trying to initialize libovsdb SB client: %v", err)
 		}
 
-		cm, err := controllerManager.NewNetworkControllerManager(ovnClientset, runMode.identity,
+		cm, err = controllerManager.NewNetworkControllerManager(ovnClientset, runMode.identity,
 			masterWatchFactory, libovsdbOvnNBClient, libovsdbOvnSBClient, eventRecorder, wg)
 		if err != nil {
 			return err
 		}
-		err = cm.Start(ctx)
-		if err != nil {
-			return fmt.Errorf("failed to start ovnkube controller: %w", err)
-		}
-		defer cm.Stop()
 
-		// record delay until ready
-		metrics.MetricOVNKubeControllerReadyDuration.Set(time.Since(startTime).Seconds())
+		// start NetworkControllerManager in a separate goroutine to allow parallel startup for NodeNetworkControllerManager.
+		// This is only required in ovnkube-controller + ovnkube-node mode, but shouldn't hurt other configs
```

Comment:
last time I tested it, these was a dependency from both sides so it was required (but now I don't remember the details)
but other than that, it also speeds up the startup (or actually doesn't regress on the startup time compared to the separate containers mode) so I thought we would need it anyway

Reply:
I am surprised though because again microshift already does this with no issue and we are changing it for them as well so it would be good to know the specifics. But just a nit anyway.

Reply:
that is interesting, so I checked
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/network-controller-manager/network_controller_manager.go#L337-L355 here ovnkube-controller waits for ovnkube-node to set an annotation https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/node_annotations.go#L86-L88

from what I see, microshit doesn't pass any zone, so it uses the default "global" zone, and there is a convenient `return true` for that case https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/node_annotations.go#L922-L924